source/conf.py

```diff
@@ -39,6 +39,8 @@
     'sphinx.ext.doctest',
 ]
 
+napoleon_include_init_with_doc = True
```

Comment:
It depends.  `UsageInputs` doesn't have a docstring.  `UsageInputs.__init__` _is_ documented.  So without this flag, the rendered docs would not include this information:

```
def __init__(self, **kwargs: typing.Any):
        """
        Parameters
        ----------
        kwargs : ExampleInputs
            Inputs to be passed in as keyword arguments to `Usage.action`.
        """
```

I could move docstrings for `__init__` methods into the docstring for the class itself instead.  Looks like this is what [numpy does](https://github.com/numpy/numpy/blob/996e6419e44668a6daa28f58efcc0d2c07b3a9da/numpy/lib/_datasource.py#L197).  See all the rendered docs for this class [here](https://numpy.readthedocs.io/en/latest/reference/generated/numpy.DataSource.html#numpy.DataSource).

Reply:
I think you might've missed my suggestion above about removing undocumented members. As it currently stands, the only fully documented member in the `usage` module (that isn't on the ignore list) is `Usage`. I think the `napoleon_include_init_with_doc` is obfuscating the fact that those other objects aren't fully documented.

Reply:
I intentionally excluded the suggestion above as well as this one because they are related to unresolved issues.  After accepting your other suggestion to remove the docstrings from `UsageInputs`, `UsageOutputNames`, and `UsageAction`, those objects no longer appear in the rendered docs without `napoleon_include_init_with_doc`.  Do you want those objects in the rendered docs? I was assuming you do, which is why I have left this and the above discussion items unresolved until we determine how to do that.

I agree that we should remove both `:undoc-members:` and `napoleon_include_init_with_doc`.  If we do want  `UsageInputs`, `UsageOutputNames`, and `UsageAction` in the rendered docs, I propose moving their `__init__` method docstrings into their main docstring, in which case they will appear in rendered docs with the removal of `:undoc-members:` and `napoleon_include_init_with_doc`.

Reply:
(resolve on a Zoom call)